pkg/manager/manager_rules.go

```
@@ -0,0 +1,318 @@
+	lock.Installed = append(lock.Installed, rule.Key)
+
+	err = lock.Save(cursorDir)
+	if err != nil {
+		return fmt.Errorf("failed to update lockfile: %w", err)
+	}
+
+	return nil
+}
+
+// RemoveRule uninstalls a rule and removes its files.
+func RemoveRule(cursorDir string, ruleKey string) error {
```

golangci-lint (1.64.8)

cognitive complexity 24 of func `RemoveRule` is high (> 20)

(gocognit)